src/metpy/plots/declarative.py

```diff
@@ -1385,19 +1473,23 @@ def _set_need_rebuild(self, _):
 
     def _build(self):
         """Build the plot by calling needed plotting methods as necessary."""
-        x, y, u, v = self.plotdata
-        if self.earth_relative:
-            transform = ccrs.PlateCarree()
+        x_like, y_like, u, v = self.plotdata
+
+        # Conditionally apply the proper transform
+        if isinstance(self.parent, MapPanel):
```

Comment:
For the `plot_kwargs` above, this could then be:
```python
if 'transform' in kwargs and self.earth_relative:
    kwargs['transform'] = ccrs.PlateCarree()
```